modules/core/src/main/java/org/apache/ignite/internal/binary/BinaryUtils.java

```diff
@@ -2253,6 +2276,38 @@ public static void writeIgniteUuid(BinaryRawWriter out, @Nullable IgniteUuid val
             return null;
     }
 
+    /**
+     * Determines whether singleton collection serialization enabled.
+     *
+     * @return {@code true} if custom Java serialization logic exists, {@code false} otherwise.
+     * @see IgniteSystemProperties#IGNITE_SUPPORT_SINGLETON_COLLECTION_SERIALIZATION .
+     */
+    public static boolean isSingletonCollectionSerializationEnabled() {
+        return Boolean.getBoolean(IgniteSystemProperties.IGNITE_SUPPORT_SINGLETON_COLLECTION_SERIALIZATION);
```

Comment:
IMO it would be a monumental slowdown to do this operation when appraising every object!

Reply:
Fixed